[SPARK-45681][UI] Clone a js version of UIUtils.errorMessageCell for consistent error rendering on UI

yaooqinn opened the pull request:
<!--
Thanks for sending a pull request!  Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
  2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'.
  4. Be sure to keep the PR description updated to reflect all changes.
  5. Please write your PR title to summarize what this PR proposes.
  6. If possible, provide a concise example to reproduce the issue for a faster review.
  7. If you want to add a new configuration, please read the guideline first for naming configurations in
     'core/src/main/scala/org/apache/spark/internal/config/ConfigEntry.scala'.
  8. If you want to add or modify an error type or message, please read the guideline first in
     'core/src/main/resources/error/README.md'.
-->

### What changes were proposed in this pull request?
<!--
Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue. 
If possible, please consider writing useful notes for better and faster reviews in your PR. See the examples below.
  1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
  2. If you fix some SQL features, you can provide some references of other DBMSes.
  3. If there is design documentation, please add the link.
  4. If there is a discussion in the mailing list, please add the link.
-->

The task table on the stage page has its own version of error rendering using js, which does not catch up with recent updates on the java side.

In this PR, we cloned a js version of UIUtils.errorMessageCell to make them behave the same

### Why are the changes needed?
<!--
Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, you can clarify why it is a bug.
-->

UI improvement

### Does this PR introduce _any_ user-facing change?
<!--
Note that it means *any* user-facing change including all aspects such as the documentation fix.
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If possible, please also clarify if this is a user-facing change compared to the released Spark versions or within the unreleased branches such as master.
If no, write 'No'.
-->

yes, errors in the task table on the stage page will be abbreviated to error class first

### How was this patch tested?
<!--
If tests were added, say they were added here. Please make sure to add some test cases that check the changes thoroughly including negative and positive cases if possible.
If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check, and descendants can verify in the future.
If tests were not added, please describe why they were not added and/or why it was difficult to add.
If benchmark tests were added, please run the benchmarks in GitHub Actions for the consistent environment, and the instructions could accord to: https://spark.apache.org/developer-tools.html#github-workflow-benchmarks.
-->

locally verified

#### before

![image](https://github.com/apache/spark/assets/8326978/bd757ced-50b5-426e-8be1-1d0a74a2fead)

#### after

![image](https://github.com/apache/spark/assets/8326978/3812aff7-6051-4f3e-b296-3af013148a3b)


### Was this patch authored or co-authored using generative AI tooling?
<!--
If generative AI tooling has been used in the process of authoring this patch, please include the
phrase: 'Generated-by: ' followed by the name of the tool and its version.
If no, write 'No'.
Please refer to the [ASF Generative Tooling Guidance](https://www.apache.org/legal/generative-tooling.html) for details.
-->

no

yaooqinn committed:
[UI] Create js version of UIUtils.errorMessageCell to for error parsing on stage page

yaooqinn committed:
Merge branch 'master' into SPARK-45681

sarutak commented on core/src/main/resources/org/apache/spark/ui/static/utils.js:
```diff
@@ -231,4 +231,42 @@ function createRESTEndPointForMiscellaneousProcess(appId) {
+  let isMultiline = true;
+  const maybeErrorClass = errorMessage.match(ERROR_CLASS_REGEX);
+  let errorClassOrBrief;
+  if (maybeErrorClass) {
```
Is it better to check the length of `maybeErrorClass` is enough?

sarutak reviewed (commented):
@yaooqinn 
Sorry for the late reply. This change almost LGTM.
Could you show the screenshots in the description to help easily understand the difference between before and after this change?

yaooqinn committed:
address comments

yaooqinn commented:
Thank you @sarutak. I updated the PR desc with UI changes before and after this PR and addressed your comments. Please take a look again when you have some time.

sarutak reviewed (approved):
LGTM. Pending GA.

dongjoon-hyun reviewed (approved):
+1, LGTM.

yaooqinn commented:
Thank you @sarutak and @dongjoon-hyun.

Test passed. Merged to master
